cli/text_encoding.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+use encoding_rs::*;
+use std::{
+  borrow::Cow,
+  io::{Error, ErrorKind},
+};
+
+/// Attempts to detect the character encoding of the provided bytes.
+///
+/// Supports UTF-8, UTF-16 Little Endian and UTF-16 Big Endian.
+pub fn detect_charset(bytes: &[u8]) -> &str {
+  const UTF8_BOM: &[u8] = b"\xEF\xBB\xBF";
+  const UTF16_LE_BOM: &[u8] = b"\xFF\xFE";
+  const UTF16_BE_BOM: &[u8] = b"\xFE\xFF";
+
+  if bytes.starts_with(UTF8_BOM) {
```

Comment:
UTF8 text is not supposed to contain a BOM, and also if it doesn't contain a UTF16 BOM then we assume UTF8 anyway, so I would suggest to remove this branch.